### PR TITLE
feat: implement election post

### PR DESCRIPTION
**Pull request by dignifiedquire**

### CODE REVIEW

File: filecoin-proofs/src/api/post.rs [hunk 1]
```diff
@@ -135,65 +144,54 @@ impl PublicReplicaInfo {
...
+    pub fn safe_comm_r(&self) -> Result<<DefaultTreeHasher as Hasher>::Domain, failure::Error> {
         as_safe_commitment(&self.comm_r, "comm_r")
     }
 }
 
-/// Generates a proof-of-spacetime.
-/// Accepts as input a challenge seed, configuration struct, and a vector of
-/// sealed sector file-path plus CommR tuples, as well as a list of faults.
-pub fn generate_post(
+/// Generates a proof-of-spacetime candidate for ElectionPoSt.
```
Comment (porcuquine) [inline filecoin-proofs/src/api/post.rs]: Should be candidates, plural.

File: storage-proofs/src/election_post.rs [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+
+    let mut hasher = Sha256::new();
+    hasher.input(&randomness[..]);
+    hasher.input(&sector_challenge_index.to_le_bytes()[..]);
+    hasher.input(&leaf_challenge_index.to_le_bytes()[..]);
+    let hash = hasher.result();
+
+    let leaf_challenge = LittleEndian::read_u64(&hash.as_ref()[..8]);
+
+    let sector_part = leaf_challenge % (sector_size / (POST_CHALLENGED_NODES * NODE_SIZE) as u64);
```
Comment (porcuquine) [inline storage-proofs/src/election_post.rs]: Can this have a better name, please? Perhaps `challenged_range_index`.

Comment (dignifiedquire) [reply storage-proofs/src/election_post.rs]: fixed

File: filecoin-proofs/src/api/post.rs (hunk 1 above)
Comment (dignifiedquire) [reply filecoin-proofs/src/api/post.rs]: fixed

**Pull request merged**